pipeline/transform.py

```diff
@@ -0,0 +1,145 @@
+            continue
+
+        if items["url"]:
+            release_date = get_release_date(items["url"])
+            genres = get_genres_from_url(items["url"])
+        else:
+            release_date = "None"
+            genres = []
+
+        sale_date = convert_from_unix_to_datetime(items["utc_date"])
+
+        event_information = {"item_type": items["item_type"],
+                             "item_description": items["item_description"],
+                             "album_title": items["album_title"],
+                             "artist_name": items["artist_name"],
+                             "country": items["country"],
+                             "amount_paid_usd": items["amount_paid_usd"],
+                             "genres": genres,
+                             "release_data": release_date,
+                             "sale_date": sale_date
+                             }
+
+        sales_info.append(event_information)
+
+    return sales_info
+
+
+def get_genres_from_url(artist_url: str) -> list[str]:
+    """Gets the lists of genres from a given artist url.
+        Returns an empty list if none can be found."""
+
+    full_url = convert_to_full_url(artist_url)
+
+    response = requests.get(full_url, timeout=1000)
+
+    if response.status_code == 200:
+        logging.info("Getting genre tags for %s", full_url)
+        soup = BeautifulSoup(response.text, 'html.parser')
+        tags = soup.find_all('a', class_='tag')
+        if tags:
+            genres = [tag.text for tag in tags]
+            return genres
+
+        logging.info("No genres listed for %s", full_url)
+        return []
+
+    logging.info("Failed to connect to %s, Status Code: %s",
+                 full_url, response.status_code)
+    return []
+
+
+def get_release_date(artist_url: str) -> str:
+    """Retrieves the release of a song or album
+    provided in the url, returns an empty string
+    if it cannot be found"""
+
+    full_url = convert_to_full_url(artist_url)
+
+    response = requests.get(full_url, timeout=1000)
+
+    if response.status_code == 200:
+        logging.info("Getting release date for for %s", full_url)
+        soup = BeautifulSoup(response.text, 'html.parser')
+        meta_tag = soup.find('meta', attrs={'name': 'description'})
+
+        if meta_tag:
+            description_content = meta_tag.get('content')
+            if description_content:
+                date_pattern = r'released\s(\d{1,2}\s[a-zA-Z]+\s\d{4})'
+                match = re.search(date_pattern, description_content)
+                if match:
+                    release_date = match.group(1)
+
+                    return release_date
+            else:
+                logging.info("No release date listed for %s", full_url)
+
+    logging.info("Failed to retrieve the webpage: %s", full_url)
+    return ""
+
+
+def create_dataframe(sales_info: list[dict]) -> pd.DataFrame:
+    """Returns a dataframe based on received sales information"""
```

Comment:
Could the docstring be "Converts sales information into a Pandas DataFrame and saves it to a CSV file."